### ISSUE TITLE
Investigate why play-pac4j-java-demo is not working with Scala 3

### ISSUE THREAD

**Issue by mkurz** [playframework/playframework#12870]
### Discussed in https://github.com/orgs/playframework/discussions/12867

<div type='discussions-op-text'>

<sup>Originally posted by **HenningHeinz** September 25, 2024</sup>
Dear Community, 

I first posted this on 
https://stackoverflow.com/questions/79019063/the-latest-pac4j-play-framework-3-java-demo-does-not-compile-start

but the error message seem to be based on a change in Play Framework (3 Java). I am using Pac4J Security library and the recent demo application release does not run / compile on my JDK 17, Linux environment, Pekko engine).

```
[NoSuchMethodError: 'void play.api.mvc.EssentialFilter.$init$(play.api.mvc.EssentialFilter)']

org.pac4j.play.filters.SecurityFilter.<init>(SecurityFilter.scala:63)
org.pac4j.play.filters.SecurityFilter$$FastClassByGuice$$c834f8.GUICE$TRAMPOLINE(<generated>)
org.pac4j.play.filters.SecurityFilter$$FastClassByGuice$$c834f8.apply(<generated>)
     com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:82)
     com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
     com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
[...]
```

2.9.5 works with pac4j-play, but complains that it is missing the jakarta.x.validation class (documented in the migration guide though).

The problem should be reproducible by compiling the demo app from github.
https://github.com/pac4j/play-pac4j-java-demo

Any help/feedback is appreciated.

Thanks in advance.

Henning</div>

**Comment by mcavigelli** [comment]
For me
```
sbt compile
sbt run
```
worked fine on Windows 11, Java 21.

The way mentioned in README.md to start it, with activator run did not work though.

**Comment by mkurz** [comment]
Yeah I can confirm it's working now (can also confirm it was broken before because locally I still was on the old commit from back then).

> The way mentioned in README.md to start it, with activator run did not work though.

- https://github.com/pac4j/play-pac4j-java-demo/pull/357

Thanks!
